### PR TITLE
Changed code to improve output for test/e2e/windows

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Enhancement

#### What this PR does / why we need it:
For better (more informative) output for developers when test fails. Changed test/e2e/windows for this PR.
#### Which issue(s) this PR fixes:

Part of #105678

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by NikhilSharmaWe** [comment]
@pohly could you please review the changes made

**Comment by NikhilSharmaWe** [comment]
@pohly could you please review the changes made

File: test/e2e/windows/cpu_limits.go [hunk 1]
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
 					found = true
 					break
 				}
-				framework.ExpectEqual(found, true, "Found pod in stats summary")
+				if !found{
+					framework.Failf("Pod with name : %v and namespace : %v not found in the stats summary %+v", p.Name, p.Namespace, allPods)
```
Comment (pohly) [inline test/e2e/windows/cpu_limits.go:86]: I'm not a fan of ` : ` as separator between name/value pairs. Also name/namespace pairs usually just get printed as `<namespace>/<name>`.

```suggestion
					framework.Failf("Pod with name : %v and namespace : %v not found in the stats summary %+v", p.Name, p.Namespace, allPods)
```
```suggestion
					framework.Failf("Pod %s/%s not found in the stats summary %+v", p.Namespace, p.Name, allPods)
```

File: test/e2e/windows/cpu_limits.go [hunk 2]
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
...
+				}
 				framework.Logf("Pod %s usage: %v", p.Name, cpuUsage)
-				framework.ExpectEqual(cpuUsage > 0, true, "Pods reported usage should be > 0")
-				framework.ExpectEqual((.5*1.05) > cpuUsage, true, "Pods reported usage should not exceed limit by >5%")
+				if !(cpuUsage > 0){
```
Comment (pohly) [inline test/e2e/windows/cpu_limits.go:89]:
```suggestion
				if cpuUsage <= 0 {
```

File: test/e2e/windows/cpu_limits.go [hunk 3]
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
...
+					framework.Failf("Pods reported usage is %v, but it should be greater than 0", cpuUsage)
```
Comment (pohly) [inline test/e2e/windows/cpu_limits.go:90]:
```suggestion
					framework.Failf("Pod %s/%s reported usage is %v, but it should be greater than 0", p.Name, p.Namespace, cpuUsage)
```

File: test/e2e/windows/cpu_limits.go [hunk 4]
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
...
+				}
+				if !((.5*1.05) > cpuUsage){
```
Comment (pohly) [inline test/e2e/windows/cpu_limits.go:92]:
```suggestion
				if cpuUsage >= .5*1.05 {
```

File: test/e2e/windows/cpu_limits.go [hunk 5]
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
...
+					framework.Failf("Pods reported usage is %v, but it should not exceed limit by >5%", cpuUsage)
```
Comment (pohly) [inline test/e2e/windows/cpu_limits.go:93]:
```suggestion
					framework.Failf("Pod %s/%s reported usage is %v, but it should not exceed limit by >5%", p.Name, p.Namespace, cpuUsage)
```

Comment (pohly) [reply test/e2e/windows/cpu_limits.go:93]: Please do the same changes also in the other file.

**Review by pohly — CHANGES_REQUESTED** [review]

**Comment by NikhilSharmaWe** [comment]
@pohly made changes according to your review.

File: test/e2e/windows/cpu_limits.go [hunk 6]
```diff
@@ -82,10 +82,16 @@ var _ = SIGDescribe("[Feature:Windows] Cpu Resources [Serial]", func() {
...
+					framework.Failf("Pod %s/%s not found in the stats summary %+v", p.Namespace, p.Name, allPods)
+				}
 				framework.Logf("Pod %s usage: %v", p.Name, cpuUsage)
-				framework.ExpectEqual(cpuUsage > 0, true, "Pods reported usage should be > 0")
-				framework.ExpectEqual((.5*1.05) > cpuUsage, true, "Pods reported usage should not exceed limit by >5%")
+				if cpuUsage <= 0 {
+					framework.Failf"Pod %s/%s reported usage is %v, but it should be greater than 0", p.Name, p.Namespace, cpuUsage)
+				}
+				if cpuUsage >= .5*1.05 {
+					framework.Failf("Pod %s/%s reported usage is %v, but it should not exceed limit by >5%", p.Name, p.Namespace, cpuUsage)
```
Comment (pohly) [inline test/e2e/windows/cpu_limits.go:93]: Wrong order of parameters.

File: test/e2e/windows/kubelet_stats.go [hunk 7]
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
 						}
 						statsChecked = statsChecked + 1
 
-						framework.ExpectEqual(*podStats.CPU.UsageCoreNanoSeconds > 0, true, "Pod stats should not report 0 cpu usage")
-						framework.ExpectEqual(*podStats.Memory.WorkingSetBytes > 0, true, "Pod stats should not report 0 bytes for memory working set ")
-
+						if *podStats.CPU.UsageCoreNanoSeconds <= 0 {
+							framework.Failf("Pod stats report cpu usage equal to %v but it should be greater than 0", *podStats.CPU.UsageCoreNanoSeconds)
```
Comment (pohly) [inline test/e2e/windows/kubelet_stats.go:81]: Include the pod namespace+name for which this fails.

File: test/e2e/windows/kubelet_stats.go [hunk 8]
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
...
+						}
+						if *podStats.CPU.UsageCoreNanoSeconds <= 0 {
+							framework.Failf("Pod stats report bytes for memory working set equal to %v but it should be greater than 0", *podStats.Memory.WorkingSetBytes)
+						}
 						for _, containerStats := range podStats.Containers {
-							framework.ExpectEqual(containerStats.Logs != nil, true, "Pod stats should have container log stats")
+							if containerStats.Logs == nil {
+								framework.Failf("Pod stats should have container log stats, but it is equal to %v", nil)
```
Comment (pohly) [inline test/e2e/windows/kubelet_stats.go:88]: Include the pod namespace+name and container name for which this fails.

File: test/e2e/windows/kubelet_stats.go [hunk 9]
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
...
+							}
 							framework.ExpectEqual(*containerStats.Logs.AvailableBytes > 0, true, "container log stats should not report 0 bytes for AvailableBytes")
+							if *containerStats.Logs.AvailableBytes <= 0 {
+								framework.Failf("container log stats report %v bytes for AvailableBytes, but it should be greater than 0", *containerStats.Logs.AvailableBytes)
```
Comment (pohly) [inline test/e2e/windows/kubelet_stats.go:92]: Same here.

File: test/e2e/windows/kubelet_stats.go [hunk 10]
```diff
@@ -94,7 +102,9 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
 
 				durationMatch := avgDurationMs <= time.Duration(10*time.Second).Milliseconds()
 				framework.Logf("Getting kubelet stats for node %v took an average of %v milliseconds over %v iterations", targetNode.Name, avgDurationMs, iterations)
-				framework.ExpectEqual(durationMatch, true, "Collecting kubelet stats should not take longer than 10 seconds")
+				if !durationMatch {
+					framework.Failf("Collecting kubelet stats take %v seconds but it should not take longer than 10 seconds", durationMatch)
```
Comment (pohly) [inline test/e2e/windows/kubelet_stats.go:106]: s/take/took/

**Review by pohly — CHANGES_REQUESTED** [review]

File: test/e2e/windows/kubelet_stats.go (hunk 8 above)
Comment (NikhilSharmaWe) [reply test/e2e/windows/kubelet_stats.go:88]: Can you confirm that in this case Pod name and namespace will be podStats.PodRef.Name and podStats.PodRef.Namespace

Comment (pohly) [reply test/e2e/windows/kubelet_stats.go:88]: If those are valid fields, then yes. I don't know this struct.

File: test/e2e/windows/kubelet_stats.go [hunk 11]
```diff
@@ -77,12 +77,20 @@ var _ = SIGDescribe("[Feature:Windows] Kubelet-Stats [Serial]", func() {
...
+						if *podStats.CPU.UsageCoreNanoSeconds <= 0 {
+							framework.Failf("Pod %s/%s stats report cpu usage equal to %v but it should be greater than 0", podStats.PodRef.Namespace, podStats.PodRef.Name,*podStats.CPU.UsageCoreNanoSeconds)
+						}
+						if *podStats.Memory.WorkingSetBytes <= 0 {
+							framework.Failf("Pod %s/%s stats report bytes for memory working set equal to %v but it should be greater than 0", podStats.PodRef.Namespace, podStats.PodRef.Name, *podStats.Memory.WorkingSetBytes)
+						}
 						for _, containerStats := range podStats.Containers {
-							framework.ExpectEqual(containerStats.Logs != nil, true, "Pod stats should have container log stats")
+							if containerStats.Logs == nil {
+								framework.Failf("Pod %s/%s stats should have container log stats, but it is equal to %v", podStats.PodRef.Namespace, podStats.PodRef.Name,nil)
```
Comment (pohly) [inline test/e2e/windows/kubelet_stats.go:88]: There's no point in passing a nil as parameter. Just put that into the message and also make it shorts ("but it is nil").

**Review by pohly — CHANGES_REQUESTED** [review]

**Comment by NikhilSharmaWe** [comment]
@pohly could you please review the changes made.

**Comment by NikhilSharmaWe** [comment]
@pohly I am unable to understand the cause error showing here 
`There are no nodes that your pod can schedule to - check your requests, tolerations, and node selectors (0/102 nodes are available: 16 Insufficient memory, 3 node(s) had taint {dedicated: greenhouse}, that the pod didn't tolerate, 99 Insufficient cpu.)`. The same is also showing in [PR](https://github.com/kubernetes/kubernetes/pull/106200) for files under autoscaling

**Comment by pohly** [comment]
You need to fix the Go syntax.

**Comment by NikhilSharmaWe** [comment]
@pohly Merged conflict appears when I try to rebase the branch.
`user@DESKTOP-ISNGT7P MINGW64 /e/githubRepos/kubernetes (betterOutputWindows)
$ git rebase upstream/master
error: could not apply 17fec5f1d6f... Changed code to improve output for test/e2e/windows
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 17fec5f1d6f... Changed code to improve output for test/e2e/windows
Auto-merging test/e2e/windows/kubelet_stats.go
CONFLICT (content): Merge conflict in test/e2e/windows/kubelet_stats.go`

**Comment by pohly** [comment]
After the merge conflict you must manually edit `test/e2e/windows/kubelet_stats.go` to resolve the conflict.

**Comment by NikhilSharmaWe** [comment]
@pohly Made another commit for resolving the commit, it should be fine now.
Just one Query is that if I try to convert the changes again in a single commit  then will the merge conflict will occur again or not, because it should be only 1 commit in a single PR.

**Comment by pohly** [comment]
```
test/e2e/windows/cpu_limits.go:90:21: syntax error: unexpected literal "Pod %s/%s reported usage is %v, but it should be greater than 0" at end of statement
```

**Comment by pohly** [comment]
There's a govet error.

**Comment by NikhilSharmaWe** [comment]
@pohly could you recommend me a code editor because when I try to save the code base of kubernetes by go format, it takes a very long time.

**Comment by NikhilSharmaWe** [comment]
@pohly by the way I changed 5% to 5 percent because it was showing govet error

**Comment by pohly** [comment]
> by the way I changed 5% to 5 percent because it was showing govet error

You can escape `%` in a format string with `%%`.

> could you recommend me a code editor because when I try to save the code base of kubernetes by go format, it takes a very long time.

Sorry, I don't know which editor will work well in your environment. But you really need to find a way to find to avoid such simple mistakes like broken formatting or code that has syntax errors before submitting a PR update. Every such update kicks of all tests jobs, which costs the project real money and resources that could be spent better.

**Comment by NikhilSharmaWe** [comment]
> > by the way I changed 5% to 5 percent because it was showing govet error
> 
> You can escape `%` in a format string with `%%`.
> 
> > could you recommend me a code editor because when I try to save the code base of kubernetes by go format, it takes a very long time.
> 
> Sorry, I don't know which editor will work well in your environment. But you really need to find a way to find to avoid such simple mistakes like broken formatting or code that has syntax errors before submitting a PR update. Every such update kicks of all tests jobs, which costs the project real money and resources that could be spent better.

I am so sorry about that, I will keep this in mind in my upcoming PR's

**Review by pohly — APPROVED** [review]
/lgtm

**Comment by pohly** [comment]
/assign @ddebroy

For approval.

**Comment by jsturtevant** [comment]
@NikhilSharmaWe thanks! Any reason not to include updates to the new metric non-serial test you added recently? 

https://github.com/kubernetes/kubernetes/blob/51b94de68f0889e4e9bd5bc7673041f1275ca175/test/e2e/windows/kubelet_stats.go#L149-L150

**Comment by NikhilSharmaWe** [comment]
@jsturtevant I think I made the changes when the changes made for adding new metric were not merged. I will be updating the PR soon.

**Pull request merged**